.eslintrc.js

```diff
@@ -1,21 +1,39 @@
 module.exports = {
   root: true,
   extends: ['@metamask/eslint-config', '@metamask/eslint-config-nodejs'],
-  env: {
-    commonjs: true,
-  },
+
   overrides: [
     {
-      files: ['test/**/*.js'],
+      files: ['**/*.test.ts'],
       extends: ['@metamask/eslint-config-jest'],
+      rules: {
+        'node/no-unpublished-require': 0,
+      },
+    },
+    {
+      files: ['*.ts'],
+      extends: ['@metamask/eslint-config-typescript'],
     },
   ],
-  rules: {
-    camelcase: [
-      'error',
-      {
-        allow: ['signTypedData_v1', 'signTypedData_v3', 'signTypedData_v4'],
+
+  settings: {
+    'import/resolver': {
+      node: {
+        extensions: ['.js', '.jsx', '.ts', '.tsx'],
       },
-    ],
+    },
+    node: {
+      tryExtensions: ['.js', '.json', '.node', '.ts'],
+    },
   },
+
+  // This is necessary to run eslint on Windows and not get a thousand CRLF errors
```

Comment:
VSCode is fine.  It's in the terminal when you run the lint commands, the output looks like the block below.  Yeah it would be nice to add to the module template, to the extension, etc.

```
C:\Depot\MetaMask\eth-simple-keyring\src\simple-keyring.ts
    1:59  error  Delete `␍`  prettier/prettier
    2:9   error  Delete `␍`  prettier/prettier
    3:15  error  Delete `␍`  prettier/prettier
    4:15  error  Delete `␍`  prettier/prettier
    5:10  error  Delete `␍`  prettier/prettier
    6:18  error  Delete `␍`  prettier/prettier
    7:19  error  Delete `␍`  prettier/prettier
    8:19  error  Delete `␍`  prettier/prettier
    9:18  error  Delete `␍`  prettier/prettier
   10:12  error  Delete `␍`  prettier/prettier
   11:27  error  Delete `␍`  prettier/prettier
   12:9   error  Delete `␍`  prettier/prettier
   13:13  error  Delete `␍`  prettier/prettier
   14:11  error  Delete `␍`  prettier/prettier
   15:26  error  Delete `␍`  prettier/prettier
   16:13  error  Delete `␍`  prettier/prettier
   17:16  error  Delete `␍`  prettier/prettier
   18:17  error  Delete `␍`  prettier/prettier
   19:24  error  Delete `␍`  prettier/prettier
   20:33  error  Delete `␍`  prettier/prettier
   21:77  error  Delete `␍`  prettier/prettier
```